Firestore/core/src/remote/serializer.h

```diff
@@ -125,12 +126,8 @@ class Serializer {
   /**
    * @brief Converts from nanopb proto to the model FieldValue format.
    */
-  // TODO(rsgowman): Once the proto is read, the only thing the reader object is
-  // used for is error handling. This seems questionable. We probably need to
-  // rework error handling. Again. But we'll defer that for now and continue
-  // just passing the reader object.
   model::FieldValue DecodeFieldValue(
-      nanopb::Reader* reader, const google_firestore_v1_Value& proto) const;
+      util::ReadContext* context, const google_firestore_v1_Value& proto) const;
```

Comment:
Optional: ReadContext can't ever be null so this could be a mutable reference rather than a pointer (this interface was created before that change to the style guide had gone through). Feel free to leave for a follow-up if you don't want to tackle that now.

You could make this less of an intrusive fix by making `Reader::context()` return a mutable reference too.

Reply:
Added a TODO.